src/main/java/machine_learning/classifiers/ensembles/AbstractEnsemble.java

```diff
@@ -896,14 +896,12 @@ public void buildClassifier(Instances data) throws Exception {
             //we need to sum the modules' reported build time as well as the weight
             //and voting definition time
             for (EnsembleModule module : modules) {
-                buildTime += module.trainResults.getBuildTimeInNanos();
-                
-                //TODO see other todo in trainModules also. Currently working under 
-                //assumption that the estimate time is already accounted for in the build
-                //time of TrainAccuracyEstimators, i.e. those classifiers that will 
-                //estimate their own accuracy during the normal course of training
-                if (!EnhancedAbstractClassifier.classifierIsEstimatingOwnPerformance(module.getClassifier()))
-                    buildTime += module.trainResults.getErrorEstimateTime();
+                if (weightingScheme.needTrainPreds || votingScheme.needTrainPreds) {
```

Comment:
Need to be careful here, buildplusestimatetime doesnt (currently) do any sensible default, if it's not explicitly set then it's just -1 (instead of e.g. default assuming that the values in build and estimate time are mutually exclusive and just summing them)

If going to rely on it solely, need to ensure that all the base classifiers actually set that value during their internal estimate

Otherwise, either here in abstract ensemble to something like if getBuildPlusEstimateTime() <= 0, buildTime += (getBuildTime()  + getEstimateTime()), or we can put that logic straight into getBuildPlusEstimateTime()  in classifierresults

Reply:
I have updated the other HIVE-COTE components to have valid buildplusestimatetime values (except PF). Given how I believe Tony wants timings to be recorded however, there isn't much of a reason to not just automatically set it if it is == -1.

I will put the check in abstract ensemble for now.